src/app/shared/navigation-focus/navigation-focus.service.ts

```diff
@@ -0,0 +1,43 @@
+import {ElementRef, Injectable} from '@angular/core';
+import {NavigationEnd, Router} from '@angular/router';
+import {filter, skip, take} from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class NavigationFocusService {
+  private navigationFocusRequests: ElementRef[] = [];
+  private skipLinkFocusRequests: ElementRef[] = [];
+
+  constructor(private router: Router) {
+    this.router.events
+      .pipe(filter(event => event instanceof NavigationEnd),
```

Comment:
Would it be helpful to make this a generic `NavigationUtils` service and publish this pipe like
```
readonly navigationEndEvents = this.router.events.pipe(filter(event => event instanceof NavigationEnd));
readonly softNavigations = this.navigationEndEvents.pipe(skip(1));
```
There are several places in the project that could use this.

Reply:
I think the hope is to pull this all of the router subscriptions under the scope of this service anyways

Reply:
Oh nevermind, I see there are a few places that use it for things other than redirecting focus